### DISCUSSION TITLE
Suppress State Toggle for a Specific Device?

### DISCUSSION THREAD

**Discussion by rshartog** [aarons22/homebridge-bond#145] [Q&A]
I am relatively new to home automation, but I'm all in now. I have HomeKit with a Lutron Caseta GW and a Zigbee GW (Conbee II). I have several devices connected via WiFi -- Ecobee thermostats, DSC/Security System/Envisakink, and myQ garage door openers. Ecobee and Lutron support HomeKit directly and everything else is integrated into HomeKit via HOOBS running on a Raspberry Pi 4B. (None of the above information is relevant to my question as far as I know, but I offer it as a method of introduction and my commitment to home automation.)

Today I installed a Bond Bridge to control two "legacy" fans with lights and my fireplace. These devices have RF remotes. I also installed a WiFi Fan/Light controller (HomeDepot #99434) that is incorporated through the Bond Home App as a "Smart by Bond" device.

I like the state toggle option for my RF-controlled fans and my fireplace, but I would prefer not to have it for my WiFi fan, because the Bond Hub knows its state and the the toggle complicates an otherwise elegant control screen in HomeKit. As far as I can tell, the current implementation is "all or nothing" for state toggles in homebridge-bond. You get state toggle on all lights or no lights -- even across different "bonds" in the configuration.

...so here is my question: Is it possible to suppress the state toggle on an individual device? If not, then is it possible to enable the state toggle per "bond" in the config? This latter solution would work for my setup because everything connected through my bond hub has an RF remote and my WiFi device has a separate "bond" in the config, but I'm not familiar enough with other Bond devices to speculate on whether it is general enough to cover most people's configurations.

I did search the topics and I could not find this question, so please pardon me if it is already a topic (and maybe provide the search string that would expose it for my edification).

--Scott

**Comment by chrismerck** [comment]
> the state toggle option

Can you clarify what you mean by this?

Are you referring to "Trust Tracked State" in the Device Settings of the Bond Home app? (This is a feature that allows per-device control over whether the Bond Bridge will suppress toggle signals to devices it thinks are already in the desired state.)

If not, I'd like to be made smart about this "state toggle option" on the HOOBS side.

**Reply by rshartog** [reply]
I'm referring to the "include_toggle_state" option in the schema. It's currently associated with the platform configuration (platform.config.include_toggle_state in the source code). I would like to associated it (or the suppression of it) with a bond or with each individual device.

**Comment by aarons22** [comment]
Yeah, it is currently an all or nothing implementation. I've considered a config setup where you can set flags per device, just haven't gotten around to it. Can you throw a feature request into the issues tab?
